src/managedwriter/writer_client.ts

```diff
@@ -0,0 +1,311 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ** This file is automatically generated by gapic-generator-typescript. **
+// ** https://github.com/googleapis/gapic-generator-typescript **
+// ** All changes to this file may be overwritten. **
+
+import '../../package.json';
+import * as gax from 'google-gax';
+import type {CallOptions, ClientOptions} from 'google-gax';
+import * as protos from '../../protos/protos';
+
+import {BigQueryWriteClient} from '../v1';
+
+/**
+ *  BigQuery Write API.
+ *
+ *  The Write API can be used to write data to BigQuery.
+ *
+ *  For supplementary information about the Write API, see:
+ *  https://cloud.google.com/bigquery/docs/write-api
+ * @class
+ * @memberof storage
+ */
+
+type StreamConnection = {
+  write_stream: WriteStream;
+  connection: gax.CancellableStream;
+};
+type streamConnectionsMap = Record<string, gax.CancellableStream>;
+type StreamConnections = {
+  connection_list: StreamConnection[];
+  connections: streamConnectionsMap;
+};
+type WriteStream = protos.google.cloud.bigquery.storage.v1.IWriteStream;
+type AppendRowResponse =
+  protos.google.cloud.bigquery.storage.v1.AppendRowsResponse;
+type AppendRowRequest =
+  protos.google.cloud.bigquery.storage.v1.IAppendRowsRequest;
+type IInt64Value = protos.google.protobuf.IInt64Value;
+type ProtoData =
+  protos.google.cloud.bigquery.storage.v1.AppendRowsRequest.IProtoData;
+
+export class WriterClient {
+  private _opts: ClientOptions | undefined;
+  private _parent: string;
+  private _writeStreamType: WriteStream['type'] = 'TYPE_UNSPECIFIED';
+  private _streamId: string;
+  private _client: BigQueryWriteClient;
+  private _connections: StreamConnections;
```

Comment:
instead of keeping the raw connection directly here, we can keep a reference to the `ManagedStream` as I comment more below.